dotCMS/src/main/java/com/dotcms/api/content/VanityUrlFactory.java

```diff
@@ -0,0 +1,119 @@
+package com.dotcms.api.content;
+
+import java.util.List;
+
+import com.dotcms.content.model.VanityUrl;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.business.DotContentletStateException;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.liferay.portal.model.User;
+
```

Comment:
This Factory does not make sense to exists, the API is just a wrapper. My suggestion is to move all logic (even cache) to the API